backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java

```diff
@@ -212,15 +212,47 @@ public LiveOktaRepository(
           "Cannot update Okta user with unrecognized username");
     }
     User user = users.single();
+    updateUser(user, userIdentity);
+
+    return getOrganizationRoleClaimsForUser(user);
+  }
+
+  private void updateUser(User user, IdentityAttributes userIdentity) {
     user.getProfile().setFirstName(userIdentity.getFirstName());
     user.getProfile().setMiddleName(userIdentity.getMiddleName());
     user.getProfile().setLastName(userIdentity.getLastName());
     // Is it our fault we don't accommodate honorific suffix? Or Okta's fault they
     // don't have regular suffix? You decide.
     user.getProfile().setHonorificSuffix(userIdentity.getSuffix());
     user.update();
+  }
 
-    return getOrganizationRoleClaimsForUser(user);
+  public void reprovisionUser(IdentityAttributes userIdentity) {
+    UserList users = _client.listUsers(userIdentity.getUsername(), null, null, null, null);
+    if (users.stream().count() == 0) {
+      throw new IllegalGraphqlArgumentException(
+          "Cannot reprovision Okta user with unrecognized username");
+    }
+    User user = users.single();
+    UserStatus userStatus = user.getStatus();
+
+    // any org user "deleted" through our api will be in SUSPENDED state
+    if (userStatus != UserStatus.SUSPENDED) {
+      throw new IllegalGraphqlArgumentException(
+          "Cannot reprovision user in unsupported state: " + userStatus);
+    }
+
+    updateUser(user, userIdentity);
+
+    // reset all MFA factors (this triggers an email to the user, which can be configured in Okta)
+    user.resetFactors();
```

Comment:
do this line and line 255 end up sending two emails to the user? this isn't included in the reactivation step by default, is it?

Reply:
Yes, unfortunately it does result in two emails.  The first is a message about their MFA being reset and the second is an account activation email.  We need them to get the latter (which could easily be disabled), but don't want them to get the former (which can only be disabled by an across-the-board removal of MFA reset emails).

The deactivate/activate cycle by itself only resets the user's password and password reset question and leaves the MFA in place.  We need to reset all of it or else we won't be lightening the load on support.